packages/toolkit/src/view/model/view-model/ModelOverview.tsx

```diff
@@ -248,40 +267,15 @@ export const ModelOverview = ({ model, modelState }: ModelOverviewProps) => {
         },
       });
 
-      toast({
-        variant: "notification-success",
-        title: "Model was triggered",
-        size: "large",
-        description:
-          "Come back later or reload the page to see if the run result is ready.",
-        action: (
-          <div className="flex flex-row gap-x-4">
-            <LinkButton
-              onClick={() => {
-                dismiss();
-              }}
-              variant="secondary"
-              size="md"
-            >
-              Dismiss
-            </LinkButton>
-            <LinkButton
-              onClick={() => {
-                window.location.reload();
-              }}
-              variant="primary"
-              size="md"
-            >
-              Reload
-            </LinkButton>
-          </div>
-        ),
-      });
-
       if (amplitudeIsInit) {
         sendAmplitudeData("trigger_model");
       }
 
+      currentOperationIdPollingData.current = {
+        ...defaultCurrentOperationIdPollingData,
+        name: data.operation.name,
+      };
```

Comment:
@orangecoloured I am curious why you choose to use ref for this data. If it's due to the infinite re-rendering then it might be the wrongly usage of reactQuery object as deps issue I mentioned above

Reply:
ref is updated out of the render loop, like a normal object, and I need this data to be updated as soon as I update it and not on the next re-render. By the time next re-render occurs all the hooks already have the fresh data.

Reply:
@orangecoloured 

> and I need this data to be updated as soon as I update it and not on the next re-render

I think this is the place I am lost. What is the scenario that you need to stay into the react life-cycle and do something in between

Reply:
One of the main reasons - so I don't have to deal with stale data. I don't want to think about the order of the renders. I want to be sure that I can really on this temp storage that keeps the currently active operation id. I want to be sure that at this moment it keeps what I think it should keep. I don't want to also keep in mind the react render cycle.

Reply:
I think it's a valid act for me, please leave the comment in the codebase. 

But I would like to address that, normally I don't think leaving react life cycle is a good thing in the long term, people will find it's a bit hard to read through the code if we use ref to dodge the natural life-cycle of React. 

In the end, the lifecycle of the react is the first and formost item to be considered when written react code

Reply:
Agree on React lifecycle. But with a complex logic like this, hooks are hard to deal with. Implementing this with a class would be much easier and cleaner.

And I think what could help us here is some kind of a long-polling request or [SSE](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events).

Reply:
Yes, I would say after we have proper streaming among our operation endpoint, this will be much easier to implement, let's hold like this way first